### PR TITLE
(WIP) fix(Scripts/MoltenCore): Ragnaros rewrite

### PR DESCRIPTION
**Pull request by sanctum32**
Fixed Geddon self cast interrupts at Armageddon cast

## Changes Proposed:
-  Ragnaros encounter rewrite

## Issues Addressed:

- Closes 

## Tests Performed:

- tested in
- 

## How to Test the Changes:

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_ragnaros.cpp [hunk 1]
```diff
@@ -285,34 +312,43 @@ class boss_ragnaros : public CreatureScript
...
+            me->HandleEmoteCommand(EMOTE_ONESHOT_EMERGE);
+
+            if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0, 0.0f, true))
             {
-                _hasSubmergedOnce = true;
+                AttackStart(target);
             }
         }
+
+        void SchedulleCombatEvents()
```
Comment (Nefertumm) [inline src/server/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_ragnaros.cpp:343]:
```suggestion
        void ScheduleCombatEvents()
```

File: src/server/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_ragnaros.cpp [hunk 2]
```diff
@@ -285,34 +312,43 @@ class boss_ragnaros : public CreatureScript
...
-                    }
-                }
+                return;
             }
 
-            _isBanished = true;
-            _emergeTimer = 90000;
-            if (!_hasSubmergedOnce)
+            events.SetPhase(PHASE_EMERGED);
+            SchedulleCombatEvents();
```
Comment (Nefertumm) [inline src/server/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_ragnaros.cpp:328]:
```suggestion
            ScheduleCombatEvents();
```

File: src/server/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_ragnaros.cpp [hunk 3]
```diff
@@ -86,61 +90,67 @@ class boss_ragnaros : public CreatureScript
...
         {
             _EnterCombat();
-            events.ScheduleEvent(EVENT_ERUPTION, 15000);
-            events.ScheduleEvent(EVENT_WRATH_OF_RAGNAROS, 30000);
-            events.ScheduleEvent(EVENT_HAND_OF_RAGNAROS, 25000);
-            events.ScheduleEvent(EVENT_LAVA_BURST, 10000);
-            events.ScheduleEvent(EVENT_MAGMA_BLAST, 2000);
-            events.ScheduleEvent(EVENT_SUBMERGE, 180000);
+            events.SetPhase(PHASE_EMERGED);
+            SchedulleCombatEvents();
```
Comment (Nefertumm) [inline src/server/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_ragnaros.cpp:141]:
```suggestion
            ScheduleCombatEvents();
```

**Review by Nefertumm — APPROVED** [review]

**Comment by Nefertumm** [comment]
Code seems ok, just one minor stuff hehe. Also, I'm a bit dumb because I didn't see this is a draft kekw

**Comment by Nyeriah** [comment]
Hey, any updates on this?

File: src/server/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_ragnaros.cpp (hunk 1 above)
Comment (sanctum32) [reply src/server/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_ragnaros.cpp:391]: applied, thanks

**Review by Nyeriah — APPROVED** [review]

**Pull request merged**